CI: iOS Tests ワークフローが main push 後に 3 連続 failure（Install iOS Simulator Runtime で "Unable to connect to simulator"）

## 症状

main ブランチへの push 後に `.github/workflows/*` の **iOS Tests** ワークフローが 3 連続で failure。全て **Install iOS Simulator Runtime** ステップで `Unable to connect to simulator` → exit code 70。

| Run ID | Commit | 日時 | 結果 |
|--------|--------|------|------|
| 24760085320 | 506f4e8 (#158) | 2026-04-22T04:24:27Z | ❌ 41s |
| 24758496043 | 5adfa2e (#156) | 2026-04-22T03:23:33Z | ❌ 39s |
| 24754398062 | (#153)      | 2026-04-22T00:54:42Z | ❌ 38s |

注: PR 側の `xcodebuild test` (actions/runs/24757228567/job/72434355751 等) は **pass** しており、コード自体に問題なし。workflow インフラ側の flaky。

## 再現ログ

```
##[group]Run xcodebuild -downloadPlatform iOS
xcodebuild -downloadPlatform iOS
shell: /bin/bash -e {0}
env:
  MD_APPLE_SDK_ROOT: /Applications/Xcode_16.3.app
##[endgroup]
Unable to connect to simulator.
Finding content...
##[error]Process completed with exit code 70.
```

## 影響

- **main CI green の担保が不能** → PR merge 後の regression detection が事実上壊れている
- マージ判断時に「push 後も通る確信」が取れない
- PR 側 CI は pass するため表面上は検知されにくい

## 原因仮説

1. GitHub Actions の macOS runner で iOS simulator runtime ダウンロードが不安定（リージョン/時間帯依存）
2. `Xcode_16.3.app` + iOS 17 simulator の組み合わせで runner image のキャッシュが破損
3. `xcodebuild -downloadPlatform iOS` の retry 設定なし

## 対策候補

- [ ] `Install iOS Simulator Runtime` ステップに `retry` / `continue-on-error` 追加
- [ ] `actions/cache` で simulator runtime をキャッシュ
- [ ] runner image のバージョン固定（`macos-14` → `macos-15` 等）
- [ ] `xcodebuild -downloadPlatform iOS` を事前 install 済み simulator 利用に切り替え

## 優先度

**P2** — コード自体は健全で PR CI は pass。main CI は運用上の可視化を失っているが、実装ブロッカーではない。着手する場合は workflow ファイル変更のみでコード修正不要。

## 関連

- PR #156 / #158 merge 後の main CI は全て同原因で fail
- PR 側の `xcodebuild test` ローカル実行は継続して pass

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -48,6 +48,7 @@ jobs:
           MAX_ATTEMPTS: '3'
           RETRY_INTERVAL: '30'
         run: |
+          set -euo pipefail
           attempt=1
           while [ "$attempt" -le "$MAX_ATTEMPTS" ]; do
             echo "Attempt $attempt of $MAX_ATTEMPTS: xcodebuild -downloadPlatform iOS"
@@ -61,16 +62,7 @@ jobs:
             fi
             attempt=$((attempt + 1))
           done
-          if xcrun simctl list runtimes --json | python3 -c "
-          import json, sys
-          d = json.load(sys.stdin)
-          ok = any('iOS' in r.get('identifier', '') and r.get('isAvailable') for r in d.get('runtimes', []))
-          sys.exit(0 if ok else 1)
-          "; then
-            echo "::warning::Download failed after $MAX_ATTEMPTS attempts, but iOS simulator runtime is already available - continuing"
-            exit 0
-          fi
-          echo "::error::Failed to install iOS simulator runtime after $MAX_ATTEMPTS attempts and no fallback runtime available"
+          echo "::error::Failed to install iOS simulator runtime after $MAX_ATTEMPTS attempts"
           exit 1
 
       - name: Install XcodeGen

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -44,7 +44,34 @@ jobs:
           xcode-version: '16.3'
 
       - name: Install iOS Simulator Runtime
-        run: xcodebuild -downloadPlatform iOS
+        env:
+          MAX_ATTEMPTS: '3'
+          RETRY_INTERVAL: '30'
+        run: |
+          attempt=1
+          while [ "$attempt" -le "$MAX_ATTEMPTS" ]; do
+            echo "Attempt $attempt of $MAX_ATTEMPTS: xcodebuild -downloadPlatform iOS"
+            if xcodebuild -downloadPlatform iOS; then
+              echo "Simulator runtime installed successfully"
+              exit 0
+            fi
+            if [ "$attempt" -lt "$MAX_ATTEMPTS" ]; then
+              echo "Attempt $attempt failed, retrying in ${RETRY_INTERVAL}s..."
+              sleep "$RETRY_INTERVAL"
+            fi
+            attempt=$((attempt + 1))
+          done
+          if xcrun simctl list runtimes --json | python3 -c "
+          import json, sys
+          d = json.load(sys.stdin)
+          ok = any('iOS' in r.get('identifier', '') and r.get('isAvailable') for r in d.get('runtimes', []))
+          sys.exit(0 if ok else 1)
+          "; then
+            echo "::warning::Download failed after $MAX_ATTEMPTS attempts, but iOS simulator runtime is already available - continuing"
+            exit 0
+          fi
+          echo "::error::Failed to install iOS simulator runtime after $MAX_ATTEMPTS attempts and no fallback runtime available"
+          exit 1
 
       - name: Install XcodeGen
         run: brew install xcodegen

